### PR TITLE
Add basic doc examples for `std::panic::{set_hook, take_hook}`.

### PR DESCRIPTION
**Pull request by frewsxcv**

### CODE REVIEW

File: src/libstd/panicking.rs [hunk 1]
```diff
@@ -84,6 +84,18 @@ static mut HOOK: Hook = Hook::Default;
...
 ///
 /// Panics if called from a panicking thread.
+///
+/// # Examples
+///
+/// ```
+/// use std::panic;
+///
+/// panic::set_hook(Box::new(|_| {
+///     println!("Caught a panic!");
```
Comment (durka) [inline src/libstd/panicking.rs]: Might want to stay away from the word "catch" here.

Comment (frewsxcv) [inline src/libstd/panicking.rs]: "Encountered"?

**Comment by frewsxcv** [comment]
@GuillaumeGomez Added an example for `take_hook`.

**Comment by GuillaumeGomez** [comment]
Euh...

```bash
Compiling rustc_metadata v0.0.0 (file:///build/src/librustc_metadata)
error[E0282]: unable to infer enough type information about `_`
   --> src/librustc_metadata/decoder.rs:358:25
    |
358 |             predicates: (0..self.read_usize()?).map(|_| {
    |                         ^ cannot infer type for `_`
    |
    = note: type annotations or generic parameter binding required

error: aborting due to previous error
```

I restart travis.

**Comment by frewsxcv** [comment]
@GuillaumeGomez Tests are passing now.

**Review by Mark-Simulacrum — APPROVED** [review]

**Comment by GuillaumeGomez** [comment]
Great, thanks!

@bors: r+ rollup

**Pull request merged**